runtime/src/append_vec.rs

```diff
@@ -187,17 +199,39 @@ impl AppendVec {
 
         let map = unsafe { MmapMut::map_mut(&data)? };
         self.map = map;
+
+        if !self.sanitize_layout_and_length() {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
```

Comment:
I know using those Errors is a bit off...

Reply:
Yea.. I would prefer using either a custom Result type or maybe even something like `io::Result::InvalidInput` https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.InvalidInput